### ISSUE TITLE
Disable colors for junit reports

### ISSUE THREAD

**Issue by gartentrio** [computationalmodelling/nbval#91]
We use pytest and nbval to generate junit-xml files. Unfortunately there is no option to disable the coloring of output messages. Here is an example junit xml file:

```
<testsuite errors="0" failures="1" name="pytest" skips="0" tests="3" time="3.044">
<testcase classname="notebooks.tests.test_error.ipynb" file="notebooks/tests/test_error.ipynb" line="0" name="Cell 0" time="1.4594926834106445">
<failure message="#x1B[91mNotebook cell execution failed#x1B[0m #x1B[94mCell 0: Cell execution caused an exception Input: #x1B[0mfoo = "Hase" assert foo == "Igel" #x1B[94mTraceback:#x1B[0m #x1B[0;31m---------------------------------------------------------------------------#x1B[0m #x1B[0;31mAssertionError#x1B[0m Traceback (most recent call last) #x1B[0;32m<ipython-input-1-24658342da6f>#x1B[0m in #x1B[0;36m<module>#x1B[0;34m()#x1B[0m #x1B[1;32m 1#x1B[0m #x1B[0mfoo#x1B[0m #x1B[0;34m=#x1B[0m #x1B[0;34m"Hase"#x1B[0m#x1B[0;34m#x1B[0m#x1B[0m #x1B[1;32m 2#x1B[0m #x1B[0;34m#x1B[0m#x1B[0m #x1B[0;32m----> 3#x1B[0;31m #x1B[0;32massert#x1B[0m #x1B[0mfoo#x1B[0m #x1B[0;34m==#x1B[0m #x1B[0;34m"Igel"#x1B[0m#x1B[0;34m#x1B[0m#x1B[0m #x1B[0m #x1B[0;31mAssertionError#x1B[0m: ">
#x1B[91mNotebook cell execution failed#x1B[0m #x1B[94mCell 0: Cell execution caused an exception Input: #x1B[0mfoo = "Hase" assert foo == "Igel" #x1B[94mTraceback:#x1B[0m #x1B[0;31m---------------------------------------------------------------------------#x1B[0m #x1B[0;31mAssertionError#x1B[0m Traceback (most recent call last) #x1B[0;32m<ipython-input-1-24658342da6f>#x1B[0m in #x1B[0;36m<module>#x1B[0;34m()#x1B[0m #x1B[1;32m 1#x1B[0m #x1B[0mfoo#x1B[0m #x1B[0;34m=#x1B[0m #x1B[0;34m"Hase"#x1B[0m#x1B[0;34m#x1B[0m#x1B[0m #x1B[1;32m 2#x1B[0m #x1B[0;34m#x1B[0m#x1B[0m #x1B[0;32m----> 3#x1B[0;31m #x1B[0;32massert#x1B[0m #x1B[0mfoo#x1B[0m #x1B[0;34m==#x1B[0m #x1B[0;34m"Igel"#x1B[0m#x1B[0;34m#x1B[0m#x1B[0m #x1B[0m #x1B[0;31mAssertionError#x1B[0m:
</failure>
</testcase>
<testcase classname="notebooks.tests.test_error.ipynb" file="notebooks/tests/test_error.ipynb" line="0" name="Cell 1" time="1.2278406620025635"/>
<testcase classname="notebooks.tests.test_error.ipynb" file="notebooks/tests/test_error.ipynb" line="0" name="Cell 2" time="0.34316015243530273"/>
</testsuite>
```

**Comment by vidartf** [comment]
While this is probably a reasonable request, a temporary workaround would be to pass the output through a regexp like this: https://stackoverflow.com/questions/14693701/how-can-i-remove-the-ansi-escape-sequences-from-a-string-in-python

**Comment by gartentrio** [comment]
Thanks, could you please provide an example where to place the code you mentioned.

**Comment by vidartf** [comment]
That's up to you, but you would necessarily have to input it somewhere between nbval and junit. Maybe you can pipe the output of nbval through a custom python script? Maybe a shell script is better for you?

**Comment by gartentrio** [comment]
As already mentioned, there is a working workaround for the problem.

**Comment by vidartf** [comment]
Actually, there is a much simpler solution (came over it by accident): add the option `--color=no` to the py.test call.

**Comment by vidartf** [comment]
No wait, that was just colors being screwed up in my terminal. I'll add some changes to make nbval respect the py.test `--color` flag.

**Comment by vidartf** [comment]
Suggested long-term fix in #93.
